improvement: CLDSRV-60 cleanup replay IDs of null versions

jonathan-gramain opened the pull request:
* cleanup replay IDs of null versions

Replace the previous way of passing replayId option to metadata DELETE by the result of the preprocessingVersioningDelete function, since it handles more cases like when a bucket with versioning suspended has to delete the previous null version, in which case, the null version's replay IDs have to be cleaned up.

* process upload IDs in versioning helpers

In versioning helpers, make sure we pass on the nullUploadId/replayId to the metadataStoreObject() or deleteObject() function, so that they can update ObjectMD and pass the appropriate replay options to metadata

jonathan-gramain committed:
improvement: CLDSRV-60 process upload IDs in versioning helpers

In versioning helpers, make sure we pass on the nullUploadId/replayId
to the metadataStoreObject() or deleteObject() function, so that they
can update ObjectMD and pass the appropriate replay options to metadata

jonathan-gramain committed:
improvement: CLDSRV-60 cleanup replay IDs of null versions

Replace the previous way of passing replayId option to metadata DELETE
by the result of the preprocessingVersioningDelete function, since it
handles more cases like when a bucket with versioning suspended has to
delete the previous null version, in which case, the null version's
replay IDs have to be cleaned up.

jonathan-gramain committed:
CLDSRV-60 bump arsenal dep

watsaqat commented on tests/unit/api/apiUtils/versioning.js:
```diff
@@ -43,6 +43,24 @@ describe('versioning helpers', () => {
                 },
             },
+            {
+                // prior MPU object non-null version exists
```
instead of commenting the description of the test case, it might be better to do:
`{ description: 'prior MPU object non-null version exists }`
and incorporate this into `it('..${ testCase.description }..'` in forEach

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.10/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions` with contents from `improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions`
and `development/7.10`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.10/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions origin/development/7.10
 $ git merge origin/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.10/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions
```

jonathan-gramain committed:
CLDSRV-60 address review: improve test self-doc

In versioning tests, add a 'description' field in test cases instead
of comments to make the description part of the test name

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.2/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions` with contents from `w/7.10/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions`
and `development/8.2`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.2/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions origin/development/8.2
 $ git merge origin/w/7.10/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions` into
`development/7.4`
* [w/7.10/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions](https://github.com/scality/cloudserver/compare/development/7.10...w/7.10/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions) will be merged into `development/7.10` (pull request #4102)
* [w/8.2/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions](https://github.com/scality/cloudserver/compare/development/8.2...w/8.2/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions) will be merged into `development/8.2` (pull request #4103)
* [w/8.3/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions](https://github.com/scality/cloudserver/compare/development/8.3...w/8.3/improvement/CLDSRV-60-cleanupReplayIdOfPreviousVersions) will be merged into `development/8.3` (pull request #4104)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

jonathan-gramain commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.2`

* :heavy_check_mark: `development/8.3`



The following branches will **NOT** be impacted:

* `development/6.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.2`

* :heavy_check_mark: `development/8.3`



The following branches have **NOT** changed:

* `development/6.4`



Please check the status of the associated issue CLDSRV-60.

Goodbye jonathan-gramain.
